pkg/testutil/testutil_test.go

```diff
@@ -0,0 +1,174 @@
+package testutil
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gitpod-io/leeway/pkg/leeway"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestLoadFromYAML(t *testing.T) {
+	ignores := cmpopts.IgnoreUnexported(
+		leeway.Workspace{},
+		leeway.Component{},
+		leeway.Package{},
+		leeway.WorkspaceProvenance{},
+		leeway.GitInfo{},
+	)
+
+	tests := []struct {
+		Name        string
+		Content     string
+		Expectation *Setup
+	}{
+		{
+			Name:        "empty",
+			Expectation: &Setup{},
+		},
+		{
+			Name: "single",
+			Expectation: &Setup{
+				Components: []Component{
+					{
+						Location: "comp1",
+						Packages: []leeway.Package{
+							{
+								PackageInternal: leeway.PackageInternal{
+									Name:    "pkg1",
+									Type:    leeway.GenericPackage,
+									Sources: []string{"**"},
+								},
+								Config: leeway.GenericPkgConfig{
+									Commands: [][]string{{"foo"}},
+								},
+							},
+						},
+					},
+				},
+			},
+			Content: `components:
+  - location: comp1
+    packages:
+      - name: pkg1
+        type: generic
+        srcs: 
+          - "**"
+        config:
+          commands: [["foo"]]`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			act, err := LoadFromYAML(bytes.NewBufferString(test.Content))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if diff := cmp.Diff(test.Expectation, act, ignores); diff != "" {
+				t.Errorf("LoadFromYAML() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestMaterialise(t *testing.T) {
+	type File struct {
+		Path   string
+		Size   int64
+		SHA256 string
+	}
+	tests := []struct {
```

Comment:
Nice to have: A test case showing that it fails with a reasonable error message if it can't write the files. E.g. what happens if you use a relative path that goes up the dir and tries to write somewhere you aren't allowed to?

Reply:
I've indeed taken the shortcut and built the expectation without the error, i.e. made the assumption that any error would represent test failure. This makes the integration of non-happy path tests harder ...